state-chain/runtime/src/lib.rs

```diff
@@ -1858,11 +1859,32 @@ impl_runtime_apis! {
 		fn cf_broker_info(
 			account_id: AccountId,
 		) -> BrokerInfo {
-			let earned_fees = Asset::all().map(|asset|
+			let earned_fees: Vec<_> = Asset::all().map(|asset|
 				(asset, AssetBalances::get_balance(&account_id, asset))
 			).collect();
-
-			BrokerInfo { earned_fees }
+			if let Some(channel_id) = pallet_cf_swapping::BrokerPrivateBtcChannels::<Runtime>::get(&account_id) {
+				let EpochKey { key, .. } = BitcoinThresholdSigner::active_epoch_key().expect("We should always have a key for the current epoch.");
+				BrokerInfo {
+					earned_fees,
+					deposit_address: if let Ok(salt) = channel_id.try_into() {
```

Comment:
I thought it is acceptable to return None if the conversion should fail here, since this should not happen anyway. We can also return an error in this case, but I don't know how valuable this is for the consumer of the API, and this means more overhead.

Reply:
Hmm.. Not sure if I agree. If this happens we definitely want to know about it instead of failing silently. I agree this should never happen though, so maybe this is a one of those cases where it's ok to `expect`? 

Btw. Maybe we can deduplicate this address-derivation code: this is very similar to what is in `cf_get_vault_swap_details`.

Reply:
Agree, I think expect makes sense.